src/plugins/tool_select.py

```diff
@@ -0,0 +1,82 @@
+# tool_select.py
+# Selection tool.
+# Copyright (c) 2015, Lennart Riecken
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program.  If not, see <http://www.gnu.org/licenses/>.
+from PySide import QtGui, QtCore
+from tool import Tool, EventData, MouseButtons, KeyModifiers, Face
+from plugin_api import register_plugin
+
+
+class SelectionTool(Tool):
+
+    def __init__(self, api):
+        super(SelectionTool, self).__init__(api)
+        # Create our action / icon
+        self.action = QtGui.QAction(
+            QtGui.QPixmap(":/images/gfx/icons/border.png"),
+            "Select", None)
+        self.action.setStatusTip("Select voxels")
+        self.action.setCheckable(True)
+        self.action.setShortcut(QtGui.QKeySequence("Ctrl+9"))
+        # Register the tool
+        self.api.register_tool(self)
+        # Area tool helper
+        self._first_target = None
+
+    def on_mouse_click(self, data):
+        shift_down = not not data.key_modifiers & QtCore.Qt.KeyboardModifier.ShiftModifier
+        if shift_down:
+            voxel = data.voxels.get(data.world_x,  data.world_y,  data.world_z)
+            if voxel != 0:
+                if data.voxels.is_selected(data.world_x,  data.world_y,  data.world_z):
+                    data.voxels.deselect(data.world_x,  data.world_y,  data.world_z)
+                else:
+                    data.voxels.select(data.world_x,  data.world_y,  data.world_z)
+        else:
+            data.voxels.clear_selection()
+    
+    # Start a drag
+    def on_drag_start(self, data):
+        voxel = data.voxels.get(data.world_x,  data.world_y,  data.world_z)
+        if voxel != 0:
+            self._first_target = (data.world_x,  data.world_y,  data.world_z)
```

Comment:
I think I should turn off the pull request comment integration from Codacy, because it's a little bit to noisy. But you can find a list of all new style issues here: https://www.codacy.com/app/chrmoritz/zoxel/pullRequest?bid=3023807&prid=147777